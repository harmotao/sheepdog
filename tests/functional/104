#!/bin/bash

# Test restart cluster for store_driver(tree)

. ./common

for i in 0 1 2 3 ; do
    _start_sheep $i
done
_wait_for_sheep 4
_cluster_format -c 3 -b tree

$DOG vdi create test 16M
_random | $DOG vdi write test 0 512
$DOG vdi snapshot test
_random | $DOG vdi write test 0 512
$DOG vdi clone -s 1 test test2
_random | $DOG vdi write test2 0 512
$DOG vdi delete test

echo "check vdi/cluster status"
$DOG vdi list | _filter_short_date
$DOG node list
$DOG cluster info | _filter_cluster_info | _filter_ip
_node_info

# restart sheepdog
echo "restart sheepdog cluster"
$DOG cluster shutdown
for i in 0 1 2 3 ; do
    _start_sheep $i
done
_wait_for_sheep 4

echo "operation should be fail"
$DOG vdi clone -s 1 test test2

echo "check vdi/cluster status again"
$DOG vdi list | _filter_short_date
$DOG node list
$DOG cluster info | _filter_cluster_info | _filter_ip
_node_info

status=0
