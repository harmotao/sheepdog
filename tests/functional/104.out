QA output created by 104
using backend tree store
check vdi/cluster status
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1   16 MB  4.0 MB  0.0 MB DATE   7c2b25      3                22
c test2        0   16 MB  4.0 MB  0.0 MB DATE   fd3815      3                22
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
   3   127.0.0.1:7003      	128          3
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [IP:7000:128, IP:7001:128, IP:7002:128, IP:7003:128]
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	1	2	0	0	1	0
1	2	3	0	0	1	0
2	1	3	0	0	1	0
3	2	1	0	0	0	0
restart sheepdog cluster
operation should be fail
Failed to create VDI test2: VDI exists already
check vdi/cluster status again
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1   16 MB  4.0 MB  0.0 MB DATE   7c2b25      3                22
c test2        0   16 MB  4.0 MB  0.0 MB DATE   fd3815      3                22
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
   3   127.0.0.1:7003      	128          3
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [IP:7000:128, IP:7001:128, IP:7002:128, IP:7003:128]
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	1	2	0	0	1	0
1	2	3	0	0	1	0
2	1	3	0	0	1	0
3	2	1	0	0	0	0
